src/components/calcite-pick-list/shared-list-logic.ts

```diff
@@ -0,0 +1,135 @@
+import { CalcitePickList } from "./calcite-pick-list";
+import { CalciteValueList } from "../calcite-value-list/calcite-value-list";
+
+function updateSelectedValuesMap(
+  selectedValuesMap: Map<string, HTMLCalcitePickListItemElement> | Map<string, HTMLCalciteValueListItemElement>,
```

Comment:
Nitpick: Maybe you could add a couple types in this file for `Map<string, HTMLCalcitePickListItemElement>` and `Map<string, HTMLCalciteValueListItemElement>` as it's repeated a lot and lengthy.